drivers/interrupt_controller/intc_plic.c

```diff
@@ -52,8 +67,11 @@ static int save_irq;
  */
 static int riscv_plic_is_edge_irq(uint32_t irq)
 {
+	const struct device *dev = get_plic_dev_from_irq(irq);
+	const struct plic_config *config = dev->config;
+
 	if (IS_ENABLED(CONFIG_PLIC_SUPPORTS_EDGE_IRQ)) {
-		volatile uint32_t *trig = (volatile uint32_t *)PLIC_EDGE_TRIG_TYPE;
+		volatile uint32_t *trig = (volatile uint32_t *)config->trig;
```

Comment:
here and everywhere a space after casting:
```suggestion
		volatile uint32_t *trig = (volatile uint32_t *) config->trig;
```
pet peeve of mine